src/hotspot/share/gc/z/zStringDedup.inline.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#ifndef SHARE_GC_Z_ZSTRINGDEDUP_INLINE_HPP
+#define SHARE_GC_Z_ZSTRINGDEDUP_INLINE_HPP
+
+#include "gc/z/zStringDedup.hpp"
+
+#include "classfile/javaClasses.inline.hpp"
+#include "gc/shared/gc_globals.hpp"
+
+inline void ZStringDedupContext::request(oop obj) {
+  if (!StringDedup::is_enabled()) {
+    // Not enabled
+    return;
+  }
+
+  if (!java_lang_String::is_instance(obj)) {
+    // Not a String object
+    return;
+  }
```

Comment:
Consider using `StringDedup::is_enabled_string(obj)`, which combines `is_enabled` and `is_instance`
into a single test.

Reply:
From what I can see, the `is_enable_string` takes a 'Klass*' and not an `oop`. So, that would require us to fetch the Klass bits, read the global `_klass_mode`, and decode the Klass*. I think I prefer the version we currently have because it optimize for the default (and most common run mode) that users are not using string dedup with ZGC.

Also, I don't think it is that important to optimize these checks now that the code is only run for promoted objects, which should be a significant lower number of objects compared to the number of objects that are visited by the marking code path.

Reply:
Good points.  Places where `is_enabled_string` are used already have the `Klass*` in hand for other reasons.